src/server.ts

```diff
@@ -24,62 +37,167 @@ export interface RequestMessage {
+  executeReactive?: ExecuteReactiveFunction;
+}
+
 export interface ServerOptions {
-  subscriptionManager: SubscriptionManager;
+  rootValue?: any;
+  schema?: GraphQLSchema;
+  executor?: Executor;
+  /**
+   * @deprecated subscriptionManager is deprecated, use executor instead
+   */
+  subscriptionManager?: SubscriptionManager;
   /**
-   * @deprecated onSubscribe is deprecated, use onRequest instead
+   * @deprecated onSubscribe is deprecated, use onOperation instead
    */
   onSubscribe?: Function;
   /**
-   * @deprecated onUnsubscribe is deprecated, use onRequestComplete instead
+   * @deprecated onUnsubscribe is deprecated, use onOperationComplete instead
    */
   onUnsubscribe?: Function;
-  onRequest?: Function;
-  onRequestComplete?: Function;
+  onOperation?: Function;
+  onOperationComplete?: Function;
   onConnect?: Function;
   onDisconnect?: Function;
   keepAlive?: number;
-  // contextValue?: any;
-  // rootValue?: any;
-  // formatResponse?: (Object) => Object;
-  // validationRules?: Array<any>;
-  // triggerGenerator?: (name: string, args: Object, context?: Object) => Array<{name: string, filter: Function}>;
+}
+
+class ExecuteAdapters {
+  public static executeFromExecute(execute: ExecuteFunction): ExecuteReactiveFunction {
+    return (schema: GraphQLSchema,
+            document: DocumentNode,
+            rootValue?: any,
+            contextValue?: any,
+            variableValues?: {[key: string]: any},
+            operationName?: string,
+    ) => ({
+      subscribe: (observer) => {
+        if (ExecuteAdapters.isASubscriptionOperation(document, operationName)) {
+          observer.error(new Error('Subscriptions are not supported'));
+        } else {
+          execute(schema, document, rootValue, contextValue, variableValues, operationName)
+            .then((result: ExecutionResult) => {
+                observer.next(result);
+                observer.complete();
+              },
+              (e) => observer.error(e));
+        }
+
+        return {
+          unsubscribe: () => { /* Promises cannot be canceled */ },
+        };
+      },
+    });
+  }
+
+  public static executeFromSubscriptionManager(subscriptionManager: SubscriptionManager): ExecuteReactiveFunction {
+    return (schema: GraphQLSchema,
+            document: DocumentNode,
+            rootValue?: any,
+            contextValue?: any,
+            variableValues?: {[key: string]: any},
+            operationName?: string,
+    ) => ({
+      subscribe: (observer) => {
+        if (!ExecuteAdapters.isASubscriptionOperation(document, operationName)) {
+          observer.error(new Error('Queries or mutations are not supported'));
+
+          return {
+            unsubscribe: () => { /* Empty unsubscribe method */ },
+          };
+        }
+
+        const callback = (error: Error, v: ExecutionResult) => {
+          if (error) {
+            // TODO  we should not pass ValidationError to callback in the future.
```

Comment:
no need this todo, at least not here, but on client side. ValidationError is also not populated here but on promise.reject.